src/Core/src/Test/ResultMockFactory.php

```diff
@@ -30,16 +30,26 @@ class ResultMockFactory
      *
      * @return Result|T
      */
-    public static function createFailing(string $class, int $code, ?string $message = null)
-    {
+    public static function createFailing(
+        string $class,
+        int $code,
+        ?string $message = null,
+        ?string $awsCode = null,
+        ?string $awsType = null
```

Comment:
Hm.. What would be the pros and cons by adding a 4th argument `$headers` instead of `$awsCode` and `$awsType`?

Reply:
Thought about it.. I did a bit of investigation and found out that some others aws apis might return additional `error` argument as array (https://docs.aws.amazon.com/storagegateway/latest/userguide/AWSStorageGatewayAPI.html#RESTErrorResponses)
So maybe youre right. Lets replace `$awsCode` and `$awsType` variables with like `array $additionalContent = []`. What do you think about it?